actionkit/api/user.py

```diff
@@ -48,6 +52,73 @@ def get_user(self, user_id):
         return {'res': res,
                 'user': res.json() if res.status_code == 200 else None}
 
+    def update_user(self, user_id, update_dict):
+        res = self.client.patch(
+            #the '/' at the end is IMPORTANT!
+            '%s/rest/v1/user/%s/' % (self.base_url, user_id),
+            data=json.dumps(update_dict))
+        return self._http_return(res)
+
+    def user_search(self, **kwargs):
+        res = ''
+        if kwargs['user_id']:
+            res = self.client.get('%s/rest/v1/user/%s/' % (self.base_url, kwargs['user_id']))
+        else:
+            search_string = '?'
+            search_array = []
+            for key in kwargs:
+                if kwargs[key]:
+                    item_search_string = key + '=' + kwargs[key]
+                    search_array.append(item_search_string)
+            amp = '&'
+            search_string += amp.join(search_array)
+            # for .... reasons... this can NOT have a trailing slash
+            res = self.client.get('%s/rest/v1/user/%s' % (self.base_url, search_string))
+        return {'status_code': res.status_code, 'user': res.json()}
+
+    def get_orders(self, user_id=None, orders_url=None):
+        if not orders_url:
+            user_url = '%s/rest/v1/user/%s/' % (self.base_url, user_id)
+            res = self.client.get(user_url)
+            user = res.json() if res.status_code == 200 else None
+            orders_url = user['orders']
+        orders_res = self.client.get('%s%s' % (self.base_url, orders_url))
+        orders = orders_res.json() if orders_res.status_code == 200 else None
+        return {'response': orders_res,
+                'orders': orders}
+
+    def get_orders_recurring(self, user_id=None, orders_recurring_url=None):
+        if not orders_recurring_url:
+            user_url = '%s/rest/v1/user/%s/' % (self.base_url, user_id)
+            res = self.client.get(user_url)
+            user = res.json() if res.status_code == 200 else None
+            orders_recurring_url = user['orderrecurrings']
+        orders_res = self.client.get('%s%s' % (self.base_url, orders_recurring_url))
+        orders = orders_res.json() if orders_res.status_code == 200 else None
+        return {'response': orders_res,
+                'orders': orders}
+
+    def add_phone(self, user_id, phone, phone_type, source='restful_api'):
```

Comment:
Weird that you can do it this way instead of POSTing to /rest/v1/phone/.  There are other places in the AK API where PATCHing doesn't update or create the related records. Anyway, why not simply POST to /rest/v1/phone/ ?

Reply:
I honestly have no clue. I tried a bunch of things that didn't work, and then I found this example in the AK API docs: https://roboticdogs.actionkit.com/docs/manual/api/rest/examples/userphone.html

I don't know why that works. I don't know where the post happens for the phone (I guess as part of the view layer action that updates the user?). But if you just post to /rest/v1/phone, it doesn't connect it with the user. I'm pretty sure it doesn't even create it; possibly phone requires user as a parent (???)

Reply:
I get a 201 when I do `api.client.post('https://act.moveon.org/rest/v1/phone/', json={'user': <user_id>, 'phone_type': 'home', 'phone': '9999999999'})` where api is an instance of any of the API classes in this library

Reply:
So...turns out when I was working on this, I had some other random thing wrong involving how I was constructing the url. ::headdesk::